docs: update the new faq (#1361)
GreptimeTeam/docs: docs/faq-and-others/faq.md

# Frequently Asked Questions

### How is GreptimeDB's performance compared to other solutions?

GreptimeDB has released v0.10.2, with functionalities set to improve progressively. For detailed TSBS test results, refer to the link [here](https://github.com/GreptimeTeam/greptimedb/tree/main/docs/benchmarks/tsbs).

### How does this compare to Loki? Is there a crate with Rust bindings available, preferably as a tracing or logging subscriber?

GreptimeDB now supports log data types and has introduced compatibility with various industry protocols in version 0.10. These include Loki Remote Write, Vector plugins, and the full range of OTLP data types (Metrics, Traces, Logs).

We plan to further refine the log engine, focusing on improving query performance and user experience. Future enhancements will include (but are not limited to) extending the functionality of GreptimeDB's log query DSL and implementing compatibility with some Elasticsearch/Loki APIs, providing users with more efficient and flexible log query capabilities.

For more information about using GreptimeDB with logs, refer to the documentation:

> - [Log Overview](https://docs.greptime.com/nightly/user-guide/logs/overview/)

Reviewer: ```suggestion
- [Log Overview](/user-guide/logs/overview.md)
```

> - [OpenTelemetry compatibility](https://docs.greptime.com/nightly/user-guide/ingest-data/for-observerbility/opentelemetry)

Reviewer: ```suggestion
- [OpenTelemetry compatibility](/user-guide/ingest-data/for-observerbility/opentelemetry.md)
```

> - [Loki protocol compatibility](https://docs.greptime.com/nightly/user-guide/ingest-data/for-observerbility/opentelemetry)

Reviewer: ```suggestion
- [Loki protocol compatibility](/user-guide/ingest-data/for-observerbility/opentelemetry.md)
```

> - [Vector compatibility](https://docs.greptime.com/nightly/user-guide/ingest-data/for-observerbility/vector)

Reviewer: ```suggestion
- [Vector compatibility](/user-guide/ingest-data/for-observerbility/vector.md)
```

Also, we are implementing PromQL natively which is frequently used with Grafana.

### How is the performance of GreptimeDB when used for non-time-series DB tables?

GreptimeDB supports SQL and can deal with non-time-series data, especially efficient for high concurrent and throughput data writing. However, we develop GreptimeDB for a specific domain (time-series scenarios), and it doesn't support transactions and can't delete data efficiently.

### Are there any retention policy? 

> GreptimeDB supports both database-level and table-level TTLs. By default, a table inherits the TTL of its database. However, if a table is assigned a specific TTL, the table-level TTL takes precedence. For details, refer to the official documentation on TTL: [TTL Syntax Documentation](/sql/create.md#syntax).

Reviewer: ```suggestion
GreptimeDB supports both database-level and table-level TTLs. By default, a table inherits the TTL of its database. However, if a table is assigned a specific TTL, the table-level TTL takes precedence. For details, refer to the official documentation on TTL: [TTL Syntax Documentation](/reference/sql/create.md#syntax).
```

> GreptimeDB supports both database-level and table-level TTLs. By default, a table inherits the TTL of its database. However, if a table is assigned a specific TTL, the table-level TTL takes precedence. For details, refer to the official documentation on TTL: [TTL Syntax Documentation](https://docs.greptime.com/reference/sql/create/#syntax).

Reviewer: ```suggestion
GreptimeDB supports both database-level and table-level TTLs. By default, a table inherits the TTL of its database. However, if a table is assigned a specific TTL, the table-level TTL takes precedence. For details, refer to the official documentation on TTL: [TTL Syntax Documentation](/sql/create.md#syntax).
```

### Where’s the name “Greptime” coming from?

Because `grep` is the most useful command line tool on \*nix platform to search data, and time means time series. So Greptime is to help everybody to search/find value in time series data.

### How do you measure the passing rate of PromQL compatibility tests? Is there any testing framework？

There’s [an issue](https://github.com/GreptimeTeam/greptimedb/issues/1042) to track the PromQL compatibility tests passing rate. It's based on Prometheus's compliance test.

### Does GreptimeDB support schemaless?

> Yes, GreptimeDB is a schemaless database without need for creating tables in advance. The table and columns will be created automatically when writing data with protocol gRPC, InfluxDB, OpentsDB, Prometheus remote write.

Reviewer 2: ```suggestion
Yes, GreptimeDB is a schemaless database without need for creating tables in advance. The table and columns will be created automatically when writing data with protocol gRPC, InfluxDB Line Protocol, OpenTSDB, Prometheus Remote Write.
```

Currently, GreptimeDB's compatibility efforts are primarily focused on the implementation of native PromQL. Going forward, we will continue to enhance compatibility with MetricQL's extended syntax.

### Can GreptimeDB be used for a large-scale internal metrics collection system similar to Fb's Gorilla or Google's Monarch, with a preference for in-memory data and high availability? Are there plans for asynchronous WAL or optional disk storage, and how is data replication handled without WAL?

GreptimeDB supports asynchronous WAL and is developing a per-table WAL toggle for more control. A tiered storage approach, starting with in-memory caching, is also in development. For data replication, data flushed to remote stores like S3 is replicated independently of WAL. The details for tiered storage are tracked in issue [db#2516](https://github.com/GreptimeTeam/greptimedb/issues/2516). A remote WAL implementation based on Apache Kafka ensures the durability of unflushed data in cluster mode.

### If I delete the database, can I use the `DROP DATABASE` command?

> Yes, the `DROP DATABASE` command has been implemented in version 0.8. You can refer to the official documentation for usage: [`Drop Database`](https://docs.greptime.com/reference/sql/drop#drop).

Reviewer: ```suggestion
Yes, the `DROP DATABASE` command has been implemented in version 0.8. You can refer to the official documentation for usage: [`Drop Database`](/reference/sql/drop.md#drop).
```

As mentioned, the cloud version offers more ready-to-use features to help you get started quickly. The core features are almost identical, especially on our dedicated plan.

### Where can I find documentation related to on-premises deployment and performance benchmark reports?

> You can find the public TSBS benchmark results [here](https://github.com/GreptimeTeam/greptimedb/tree/main/docs/benchmarks/tsbs) and the deployment documentation [here](https://docs.greptime.com/getting-started/installation/overview/).

Reviewer: ```suggestion
You can find the public TSBS benchmark results [here](https://github.com/GreptimeTeam/greptimedb/tree/main/docs/benchmarks/tsbs) and the deployment documentation [here](/getting-started/installation/overview.md).
```

We have prebuilt binaries for Android ARM64 platforms, which have been successfully used in some enterprise projects. However, these binaries are not available for bare metal devices, as some fundamental components of GreptimeDB require a standard library.

### Is there a built-in SQL command like `compaction table t1` that can be used for manual compaction?

> Please refer [here](/reference/sql/admin).

Reviewer: ```suggestion
Please refer [here](/reference/sql/admin.md).
```

> Please refer in this issue: [db#3363](https://github.com/GreptimeTeam/greptimedb/pull/3363).

Reviewer 2: ```suggestion
Please refer [here](/reference/sql/admin).
```

### Can GreptimeDB be used to store logs?

> Yes, please refer to this documentation for detailed information. https://docs.greptime.com/zh/user-guide/logs/overview/ 

Reviewer 2: ```suggestion
Yes, please refer [here](/user-guide/logs/overview/ ) for detailed information.
```

### How is the query performance for non-primary key fields? Can inverted indexes be set? Will the storage cost be lower compared to Elasticsearch?

Currently, non-primary key fields (or non-tag fields) do not have default inverted indexes, and we have not yet provided a `CREATE INDEX` syntax. Inverted index support will be released in an upcoming iteration along with full-text indexing. Without indexes, queries rely on MPP brute-force scanning. Although there is some parallel processing, the efficiency may not be optimal.

> As for storage costs, they will certainly be lower. You can use containers and object storage directly without relying on disks, using small local disks for buffering/caching to speed up performance. GreptimeDB employs a tiered storage architecture. For more details, please refer to our documentation on [architecture](https://docs.greptime.cn/user-guide/concepts/architecture) and [storage location](https://docs.greptime.cn/en/user-guide/concepts/storage-location).

Reviewer 2: ```suggestion
As for storage costs, they will certainly be lower. You can use containers and object storage directly without relying on disks, using small local disks for buffering/caching to speed up performance. GreptimeDB employs a tiered storage architecture. For more details, please refer to our documentation on [architecture](/user-guide/concepts/architecture) and [storage location](/en/user-guide/concepts/storage-location).
```

For more general advice for deployment, please read [Capacity Plan](/user-guide/administration/capacity-plan.md).

### In the latest version, does the Flow Engine (pre-computation) feature support PromQL syntax for calculations?

This is a good suggestion. Currently, the Flow Engine does not support PromQL syntax for calculations. We will evaluate this, as it seems theoretically feasible.

### Will Metasrv support storage backends like MySQL or PostgreSQL?

> The latest version of GreptimeDB now supports PostgreSQL as the storage backend for Metasrv. For details, please refer to [here](https://docs.greptime.com/user-guide/deployments/configuration#metasrv-only-configuration).

Reviewer 2: ```suggestion
The latest version of GreptimeDB now supports PostgreSQL as the storage backend for Metasrv. For details, please refer to [here](/user-guide/deployments/configuration#metasrv-only-configuration).
```
